compiler/noirc_evaluator/src/ssa/opt/constant_folding/mod.rs

```diff
@@ -254,7 +292,10 @@ impl Context {
             self.cached_instruction_results.get(dfg, dom, id, &instruction, predicate, block)
         {
             match cache_result {
-                CacheResult::Cached(cached) => {
+                CacheResult::Cached { instruction_id, .. } if instruction_id == id => {
```

Comment:
This looks to me like the only place where the `instruction_id` in the `CacheResult` is used. Constant folding is already one of the more memory heavy passes, does getting rid of storing the `instruction_id` make a difference? We could also look at this in a follow-up as I don't see degradations in our benchmarks. It would just be nice to avoid storing an `instruction_id` in every cache entry for what is essentially a debug assertion.

Reply:
I needed this when we had the cache during revisits, but you're right, now it's just a debug assertion, it can be removed.